tests/integration/workflows/test_views.py

```diff
@@ -90,3 +91,34 @@ def test_inspect_merge_view_returns_400(workflow_app):
     with workflow_app.test_client() as client:
         response = client.get('/workflows/inspect_merge/{}'.format(obj.id))
         assert response.status_code == 400
+
+
+def test_responses_with_etag(workflow_app):
+
+    factory = TestRecordMetadata.create_from_kwargs(
+        json={'titles': [{'title': 'Etag version'}]}
+    )
+
+    obj = workflow_object_class.create(
+        data=factory.record_metadata.json,
+        data_type='hep',
+    )
+    obj.save()
+    db.session.commit()
+
+    workflow_url = '/api/holdingpen/{}'.format(obj.id)
+
+    with workflow_app.test_client() as client:
+        login_user_via_session(client, email='cataloger@inspirehep.net')
+        response = client.get(workflow_url)
+        assert response.status_code == 200
+
+        etag = response.headers['ETag']
+        last_modified = response.headers['Last-Modified']
+
+        response = client.get(
+            workflow_url, headers={'If-Modified-Since': last_modified})
+        assert response.status_code == 304
+
+        response = client.get(workflow_url, headers={'If-None-Match': etag})
+        assert response.status_code == 304
```

Comment:
Can you add also one passing the headers but that will return the object?

Reply:
do you mean to check the object? on ``304``?

Reply:
no, I mean sending an etag different to the one that's there, and getting 200 with the new object.

Reply:
good one, done 👍